### PR TITLE
Ability to add file description on upload

### PR DESCRIPTION
**Pull request by carycheng**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @carycheng, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: boxsdk/object/folder.py [hunk 1]
```diff
@@ -263,6 +264,10 @@ def upload_stream(
             The name to give the file on Box.
         :type file_name:
             `unicode`
+        :param file_description:
+            The description to give the file on Box.
+        :type file_description:
+            `unicode`
```
Comment (mattwiller) [inline boxsdk/object/folder.py]: Should be `unicode` or None to account for the default value

File: boxsdk/object/folder.py [hunk 2]
```diff
@@ -332,6 +339,10 @@ def upload(
             The name to give the file on Box. If None, then use the leaf name of file_path
         :type file_name:
             `unicode`
+        :param file_description:
+            The description to give the file on Box. If None, then no description will be set.
+        :type file_description:
+            `unicode`
```
Comment (mattwiller) [inline boxsdk/object/folder.py]: Should be `unicode` or None to account for the default value

File: docs/usage/files.md [hunk 3]
```diff
@@ -147,7 +147,7 @@ print('File "{0}" uploaded to Box with file ID {1}'.format(new_file.name, new_fi
 ```
 
 To upload a file from a readable stream, call
-[`folder.upload_stream(file_stream, file_name, preflight_check=False, preflight_expected_size=0)`][upload_stream]
+[`folder.upload_stream(file_stream, file_name, file_description, preflight_check=False, preflight_expected_size=0)`][upload_stream]
```
Comment (mattwiller) [inline docs/usage/files.md]: The new param should have `=None` to show that it isn't required

**Pull request merged**